hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NetworkTopologyImpl.java

```diff
@@ -136,7 +137,7 @@ public void add(Node node) {
   @Override
   public void update(Node oldNode, Node newNode) {
     Preconditions.checkArgument(newNode != null, "newNode cannot be null");
-    if (oldNode != null && oldNode instanceof InnerNode) {
+    if (oldNode instanceof InnerNode) {
```

Comment:
Nice, I did not know you could avoid a null check here:

https://stackoverflow.com/questions/2950319/is-null-check-needed-before-calling-instanceof